### PR TITLE
Allow showing help in editor

### LINKED ISSUE #5730: Help documents are nowhere to be found in the web editor
People solving any exercise in the web editor don't see the content of `<track>/exercises/shared/.docs/help.md` anywhere or the global help document (can't find it ATM).

When you download the same exercise via the CLI, you get a `HELP.md` document that consists of a few different files, including those two.

As a result, a user in the web editor has no idea what to do when they're stuck, which can happen. They cannot ask for mentoring until the tests pass.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
This PR adds an icon to the editor that shows a tracks help (as defined in `exercises/shared/.docs.help.md`).

Closes https://github.com/exercism/exercism/issues/5730

change in UI:

Screenshot 2023-08-22 at 16 43 30

edit: this is what the UI looks like right now

image

### CODE REVIEW

File: app/css/pages/editor.css [hunk 1]
```diff
@@ -93,6 +94,15 @@
             }
         }
 
+        & .help-btn {
+            @apply px-6;
+            & .c-icon {
+                filter: var(--orange-filter);
```
Comment (ErikSchierboom) [inline app/css/pages/editor.css:100]: TODO: choose the right color

File: app/helpers/react_components/editor.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ def to_s
           },
           track: {
             title: track.title,
-            slug: track.slug
+            slug: track.slug,
+            help_html: Markdown::Parse.(track.git.help)
```
Comment (ErikSchierboom) [inline app/helpers/react_components/editor.rb]: What we actually want to show is an interesting question. At the least, we want to show the `exercises/shared/.docs/help.md` file, but we could also show the more general help information that we append to the `HELP.md` when an exercise is downloaded via the CLI.

For example, this is what the corresponding `HELP.md` section looks like for a C# exercise:

```markdown
## Need to get help?

If you'd like help solving the exercise, check the following pages:

- The [C# track's documentation](https://exercism.org/docs/tracks/csharp)
- [Exercism's support channel on gitter](https://gitter.im/exercism/support)
- The [Frequently Asked Questions](https://exercism.org/docs/using/faqs)

Should those resources not suffice, you could submit your (incomplete) solution to request mentoring.

To get help if you're having trouble, you can use one of the following resources:

- [Gitter](https://gitter.im/exercism/xcsharp) is Exercism C# track's Gitter room; go here to get support and ask questions related to the C# track.
- [/r/csharp](https://www.reddit.com/r/csharp) is the C# subreddit.
- [StackOverflow](http://stackoverflow.com/questions/tagged/c%23) can be used to search for your problem and see if it has been answered already. You can also ask and answer questions.
```

What's interesting is that we show the general hints first, whereas I think we should actually be showing the track-specific hints first. Thoughts?

File: app/javascript/components/editor/Header.tsx [hunk 3]
```diff
@@ -50,5 +51,5 @@ Header.ActionKeyboardShortcuts = forwardRef<
 })
 
 Header.ActionSettings = Settings
-
-Header.ActionMore = ActionMore
+Header.ActionMore = More
```
Comment (ErikSchierboom) [inline app/javascript/components/editor/Header.tsx:52]: I renamed the component from `ActionMore` to `More` to better fit with the naming of the other action components.

File: app/helpers/react_components/editor.rb (hunk 2 above)
Comment (iHiD) [reply app/helpers/react_components/editor.rb]: Inside the editor it feels like the track-specific ones are more useful. Inside the CLI it feels like the general ones are more useful.

Comment (ErikSchierboom) [reply app/helpers/react_components/editor.rb]: Okay, I've updated. This still needs some UI work, getting its own icon.

File: config/locales/exercises/en.yml [hunk 4]
```diff
@@ -4,8 +4,11 @@ en:
...
-      help_intro: "Need to get help?"
-      help: "If you'd like help solving the exercise, check the following pages:\n\n- The [%{track_title} track's documentation](https://exercism.org/docs/tracks/%{track_slug})\n- [Exercism's programming category on the forum](https://forum.exercism.org/c/programming/5)\n- The [Frequently Asked Questions](https://exercism.org/docs/using/faqs)\n\nShould those resources not suffice, you could submit your (incomplete) solution to request mentoring."
+      help_header: "Need to get help?"
+      help_intro: "If you'd like help solving the exercise, check the following pages:"
+      help_check_pages: "Should those resources not suffice, check the following pages:"
+      help_pages: "- The [%{track_title} track's documentation](https://exercism.org/docs/tracks/%{track_slug})\n- The [%{track_title} track's programming category on the forum](https://forum.exercism.org/c/programming/%{track_slug})\n- [Exercism's programming category on the forum](https://forum.exercism.org/c/programming/5)\n- The [Frequently Asked Questions](https://exercism.org/docs/using/faqs)"
```
Comment (ErikSchierboom) [inline config/locales/exercises/en.yml:13]: It gets lost in this diff, but I've added an additional resource to link to for help: the track's forum category. You can see what that looks like in some of t the tests that are modified in this PR

**Review by kytrinyx — APPROVED** [review]

**Review by kytrinyx — APPROVED** [review]

**Review by kytrinyx — APPROVED** [review]

**Review by kytrinyx — APPROVED** [review]

**Review by dem4ron — APPROVED** [review]

**Pull request merged**